type assertion with struct on valueT not working properly

The following program `sample.go` triggers a panic:

```go
package main

import (
	"errors"
	"fmt"
	"sync/atomic"
)

type wrappedError struct {
	wrapped error
}

func (e wrappedError) Error() string {
	return "some outer error"
}

func (e wrappedError) Unwrap() error {
	return e.wrapped
}

var err atomic.Value

func getWrapped() *wrappedError {
	if v := err.Load(); v != nil {
		err := v.(wrappedError)
		if err.wrapped != nil {
			return &err
		}
	}
	return nil
}

func main() {
	err.Store(wrappedError{wrapped: errors.New("test")})

	e := getWrapped()
	if e != nil {
		println(e.Error())
		println(e.wrapped.Error())
	}
}
```

Expected result:
```console
$ go run ./sample.go
some outer error
test
```

Got:
```console
$ yaegi ./sample.go
_test/type25.go:26:6: undefined field or method: wrapped
```

--- a/_test/type25.go
+++ b/_test/type25.go
@@ -1,0 +1,44 @@
+package main
+
+import (
+	"errors"
+	"sync/atomic"
+)
+
+type wrappedError struct {
+	wrapped error
+}
+
+func (e wrappedError) Error() string {
+	return "some outer error"
+}
+
+func (e wrappedError) Unwrap() error {
+	return e.wrapped
+}
+
+var err atomic.Value
+
+func getWrapped() *wrappedError {
+	if v := err.Load(); v != nil {
+		err := v.(wrappedError)
+		if err.wrapped != nil {
+			return &err
+		}
+	}
+	return nil
+}
+
+func main() {
+	err.Store(wrappedError{wrapped: errors.New("test")})
+
+	e := getWrapped()
+	if e != nil {
+		println(e.Error())
+		println(e.wrapped.Error())
+	}
+}
+
+// Output:
+// some outer error
+// test

--- a/_test/type26.go
+++ b/_test/type26.go
@@ -1,0 +1,44 @@
+package main
+
+import (
+	"errors"
+	"sync/atomic"
+)
+
+type wrappedError struct {
+	wrapped error
+}
+
+func (e *wrappedError) Error() string {
+	return "some outer error"
+}
+
+func (e *wrappedError) Unwrap() error {
+	return e.wrapped
+}
+
+var err atomic.Value
+
+func getWrapped() *wrappedError {
+	if v := err.Load(); v != nil {
+		err := v.(*wrappedError)
+		if err.wrapped != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func main() {
+	err.Store(&wrappedError{wrapped: errors.New("test")})
+
+	e := getWrapped()
+	if e != nil {
+		println(e.Error())
+		println(e.wrapped.Error())
+	}
+}
+
+// Output:
+// some outer error
+// test

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1715,7 +1715,7 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 					}
 				}
 				if n.anc.action != aAssignX {
-					if n.child[0].typ.cat == valueT && n.child[1].typ.cat != structT {
+					if n.child[0].typ.cat == valueT && !isStruct(n.child[1].typ) {
 						// Avoid special wrapping of interfaces and func types.
 						n.typ = &itype{cat: valueT, rtype: n.child[1].typ.TypeOf()}
 					} else {

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1715,7 +1715,7 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 					}
 				}
 				if n.anc.action != aAssignX {
-					if n.child[0].typ.cat == valueT {
+					if n.child[0].typ.cat == valueT && n.child[1].typ.cat != structT {
 						// Avoid special wrapping of interfaces and func types.
 						n.typ = &itype{cat: valueT, rtype: n.child[1].typ.TypeOf()}
 					} else {

